llvm-cov: Merge records for template instantiations

chapuni opened the pull request:
#119282 and #119299 will be resolved.

- #121196 
  - #121194 
    - #121192 
      - #120842 
      - #121188 
      - #121191
        - #121189 
  - #121195 
    - #110966 
      - #120841 
    - #121188 
    - #121190

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop' into users/chapuni/cov/single/trunk

chapuni committed:
Fix wrong merge resolutions

chapuni committed:
Reorganize CoverageMapping::SingleByteCoverage

chapuni committed:
Prune commented-out line

chapuni committed:
[Coverage] Make `MCDCRecord::Folded` as `[false/true]` with BitVector. NFC.

chapuni committed:
[Coverage] MCDC: Move `findIndependencePairs` into `MCDCRecord`

chapuni committed:
Merge branches 'users/chapuni/cov/merge/mcdcfold' and 'users/chapuni/cov/merge/mov_ind' into users/chapuni/cov/merge/mcdcsort-base

chapuni committed:
[Coverage] Sort `MCDCRecord::ExecVectors` order by Bitmap index

chapuni committed:
llvm-cov: Refactor CoverageSummaryInfo. NFC.

chapuni committed:
llvm-cov: Emit RegionCoverage based on `CoverageData::Segments`

chapuni committed:
Merge remote-tracking branches 'origin/users/chapuni/cov/single/refactor' and 'origin/users/chapuni/cov/single/unify' into users/chapuni/cov/merge/forfile-base

chapuni committed:
Merge branches 'users/chapuni/cov/merge/mov_ind' and 'users/chapuni/cov/merge/region_segment' into users/chapuni/cov/merge/forfile-base

chapuni committed:
llvm-cov: Use `getCoverageForFile()`

chapuni committed:
llvm-cov: Introduce `--merge-instantiations=<MergeStrategy>`

chapuni committed:
Merge branches 'users/chapuni/cov/merge/strategy' and 'users/chapuni/cov/merge/mcdcsort' into users/chapuni/cov/merge/merge-mcdc-base

chapuni committed:
llvm-cov: [MCDC] Merge and recalculate independence pairs on template instantiations.

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/refactor

Conflicts:
	llvm/test/tools/llvm-cov/branch-macros.test
	llvm/test/tools/llvm-cov/showLineExecutionCounts.test
	llvm/tools/llvm-cov/CodeCoverage.cpp
	llvm/tools/llvm-cov/SourceCoverageView.h

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/refactor

chapuni committed:
Merge branch 'users/chapuni/cov/single/refactor' into users/chapuni/cov/single/unify

chapuni committed:
Merge branch 'users/chapuni/cov/single/refactor' into users/chapuni/cov/single/nextcount

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/base

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/trunk

Conflicts:
	llvm/test/tools/llvm-cov/branch-macros.test

chapuni committed:
Merge branch 'users/chapuni/cov/single/refactor' into users/chapuni/cov/merge/trunk

Conflicts:
	llvm/test/tools/llvm-cov/branch-macros.test

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/summaryinfo

chapuni committed:
Merge branch 'users/chapuni/cov/merge/summaryinfo' into users/chapuni/cov/merge/region_segment

chapuni committed:
Merge branch 'users/chapuni/cov/merge/summaryinfo' into users/chapuni/cov/merge/mcdcsort-base

chapuni committed:
Merge branch 'users/chapuni/cov/merge/mcdcsort-base' into users/chapuni/cov/merge/trunk

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/base

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/switch

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/loop

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/condop

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/if

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/binop-base

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/loop' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/condop' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

Conflicts:
	llvm/test/tools/llvm-cov/branch-macros.test

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branches 'users/chapuni/cov/single/switch' and 'users/chapuni/cov/single/binop' into users/chapuni/cov/single/trunk

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp

chapuni committed:
s/IsSummaryEmpty/isSummaryEmpty/

chapuni committed:
Add a test

chapuni committed:
Prune redundant logic

chapuni committed:
Merge branch 'users/chapuni/cov/single/replace' into users/chapuni/cov/single/nextcount-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount-base' into users/chapuni/cov/single/nextcount

chapuni committed:
Expand RHS of MapToExpand. This will prevent recursion.

chapuni committed:
Append an explanation in the comment

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/binop-base

chapuni committed:
Update test

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/mov_ind

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/mcdcsort-base

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/unify

chapuni committed:
Merge branch 'users/chapuni/cov/single/unify' into users/chapuni/cov/merge/mcdcsort-base

chapuni committed:
Merge branch 'users/chapuni/cov/merge/mcdcsort-base' into users/chapuni/cov/merge/mcdcsort

chapuni committed:
Merge branch 'users/chapuni/cov/merge/mov_ind' into users/chapuni/cov/merge/region_segment

chapuni committed:
Merge branch 'users/chapuni/cov/merge/mov_ind' into users/chapuni/cov/single/unify

chapuni committed:
Merge branch 'users/chapuni/cov/single/unify' into users/chapuni/cov/merge/forfile-base

chapuni committed:
Merge branch 'users/chapuni/cov/merge/region_segment' into users/chapuni/cov/merge/forfile-base

chapuni committed:
Merge branch 'users/chapuni/cov/merge/forfile-base' into users/chapuni/cov/merge/forfile

chapuni committed:
Merge branch 'users/chapuni/cov/merge/mov_ind' into users/chapuni/cov/merge/trunk

chapuni committed:
Merge branch 'users/chapuni/cov/single/merge' into users/chapuni/cov/merge/strategy

chapuni committed:
Merge branch 'users/chapuni/cov/merge/forfile' into users/chapuni/cov/merge/strategy

Conflicts:
	llvm/test/tools/llvm-cov/branch-macros.test

chapuni committed:
Revert "Merge branch 'users/chapuni/cov/single/unify' into users/chapuni/cov/merge/mcdcsort-base"

This reverts commit c33e8987289f2775792977ee279c4e458f8091d5, reversing
changes made to ad6726dbfac5074c856f41adadb3b5bc14580311.

chapuni committed:
Merge branch 'users/chapuni/cov/merge/mcdcsort' into users/chapuni/cov/merge/merge-mcdc-base

chapuni committed:
Merge branch 'users/chapuni/cov/merge/strategy' into users/chapuni/cov/merge/merge-mcdc-base

chapuni committed:
Merge branch 'users/chapuni/cov/merge/merge-mcdc-base' into users/chapuni/cov/merge/merge-mcdc

chapuni committed:
Merge branch 'users/chapuni/cov/merge/merge-mcdc' into users/chapuni/cov/merge/trunk

chapuni committed:
Rewind switch DefaultCase. (to #113112)

chapuni committed:
Enable addCounters

chapuni committed:
Get rid of structual bindings

chapuni committed:
Flatten with getBranchCounterPair(SkipCntForOld)

chapuni committed:
Merge branch 'users/chapuni/cov/single/getpair' into users/chapuni/cov/single/nextcount-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount-base' into users/chapuni/cov/single/nextcount

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/base

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/switch

chapuni committed:
Resurrect merge conflict

chapuni committed:
Revert "Rewind switch DefaultCase. (to #113112)"

This reverts commit d854fb123be0449e83093f11e9f826c9baa3b766.

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/loop

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/condop

chapuni committed:
Dissolve assert

chapuni committed:
Dissolve assert

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/if

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/binop-base

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branches 'users/chapuni/cov/single/loop', 'users/chapuni/cov/single/condop' and 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/trunk

chapuni committed:
Update comments

chapuni committed:
Decorate the mock

chapuni committed:
Dissolve the old behavior

chapuni committed:
Will be pruned after the migration of SingleByte.

chapuni committed:
Add comments

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/replace

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/getpair

chapuni committed:
Merge branch 'users/chapuni/cov/single/replace' into users/chapuni/cov/single/getpair

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/pair

chapuni committed:
Merge branch 'users/chapuni/cov/single/getpair' into users/chapuni/cov/single/pair

chapuni committed:
Merge branch 'users/chapuni/cov/single/pair' into users/chapuni/cov/single/nextcount-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount-base' into users/chapuni/cov/single/nextcount

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/base

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/loop

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/condop

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/if

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/base' into users/chapuni/cov/single/switch

chapuni committed:
Merge branches 'users/chapuni/cov/single/loop', 'users/chapuni/cov/single/condop' and 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop' into users/chapuni/cov/single/trunk

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/mcdcsort

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/region_segment

chapuni committed:
Move mcdc-templates-merge.test from #121194

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/unify

chapuni committed:
Merge branch 'users/chapuni/cov/merge/region_segment' into users/chapuni/cov/merge/forfile-base

chapuni committed:
Merge branch 'users/chapuni/cov/merge/forfile-base' into users/chapuni/cov/merge/forfile

chapuni committed:
Update tests

chapuni committed:
Merge branch 'users/chapuni/cov/merge/forfile' into users/chapuni/cov/merge/strategy

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/mcdc-templates-merge.cpp
	llvm/test/tools/llvm-cov/Inputs/mcdc-templates-merge.yaml
	llvm/test/tools/llvm-cov/branch-macros.test
	llvm/test/tools/llvm-cov/mcdc-templates-merge.test

chapuni committed:
Merge branch 'users/chapuni/cov/merge/strategy' into users/chapuni/cov/merge/merge-mcdc-base

chapuni committed:
Merge branch 'users/chapuni/cov/merge/merge-mcdc-base' into users/chapuni/cov/merge/merge-mcdc

chapuni committed:
Merge branch 'users/chapuni/cov/merge/merge-mcdc' into users/chapuni/cov/merge/trunk

chapuni committed:
<numeric> std::iota

chapuni committed:
Merge branch 'users/chapuni/cov/merge/strategy' into users/chapuni/cov/merge/merge-mcdc-base

chapuni committed:
Merge branch 'users/chapuni/cov/merge/merge-mcdc-base' into users/chapuni/cov/merge/merge-mcdc

chapuni committed:
Merge branch 'users/chapuni/cov/merge/strategy' into users/chapuni/cov/merge/trunk

chapuni committed:
[MC/DC] Update CoverageMapping tests

To resolve the error, rename mcdc-error-nests.cpp -> mcdc-nested-expr.cpp at first.

- `func_condop`
  A corner case that contains close decisions.
- `func_expect`
  Uses `__builtin_expect`. (#124565)
- `func_lnot`
  Contains logical not(s) `!` among MC/DC binary operators. (#124563)

mcdc-single-cond.cpp is for #95336.

chapuni committed:
[MC/DC] Refactor MCDC::State::Decision. NFC.

Introduce `ID` and `InvalidID`. Then `DecisionByStmt` can have three
states.

* Not assigned if the Stmt(Expr) doesn't exist.
* When `DecisionByStmt[Expr]` exists:
  * Invalid and should be ignored if `ID == Invalid`.
  * Valid if `ID != Invalid`. Other member will be filled in the
    Mapper.

chapuni committed:
[MC/DC] Refactor MCDCCoverageBuilder. NFC.

- Get rid of the old `DecisionStack` and dissolve it into push/pop
  `CurCondIDs` in `VisitBin`, since `VisitBin` is recursive.

- Introduce the new `DecisionStack` with `DecisionState` to handle the
  current `Decision` in nested `Decision`s.
  - The stack has the sentinel that has `DecisionExpr = nullptr`.
  - Split out `checkDecisionRootOrPush` from `pushAndAssignIDs` for
    non-BinOp. It assigns `CondID` to `E` (instead of assignment LHS
    in `pushAndAssignIDs`).
  - The stack is manupilated at the top Decision operator in `VisitBin`.
    - The stack grows at the entrance of the Decision with the initial
      state.
    - In the same level in `VisitBin`, the stack is popped and the
      `Decision` record is emitted.
- Introduce `DecisionEndToSince` to sweep `MCDCBranch`es partially in
  `cancelDecision`.

chapuni committed:
[MC/DC] Prune MCDCLogOpStack and use CGF.isMCDCDecisionExpr

`MCDCLogOpStack` is used only for detection of the Decision root. It
can be detected with `MCDC::State::DecisionByStmt`.

chapuni committed:
[MC/DC] Create dedicated MCDCCondBitmapAddr for each Decision

MCDCCondBitmapAddr is moved from `CodeGenFunction` into `MCDCState`
and created for each Decision.

In `maybeCreateMCDCCondBitmap`, Allocate bitmaps for all valid
Decisions and emit them order by ID, to prevent nondeterminism.

chapuni committed:
[MC/DC] Handle __builtin_expect as if parenthses

Fixes #124565

chapuni committed:
[MC/DC] Enable usage of `!` among `&&` and `||`

In the current implementation, `!(a || b) && c` was not treated as one
Decision with three terms.

Fixes #124563

chapuni committed:
Update ReleaseNotes

chapuni committed:
isValid()

chapuni committed:
Remove mcdc-single-cond.cpp

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/tests

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/tests' into users/chapuni/mcdc/nest/expect

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/tests' into users/chapuni/mcdc/nest/lnot

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/tests' into users/chapuni/mcdc/nest/mcdcstate

chapuni committed:
Prune mcdc-single-cond.cpp

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/tests' into users/chapuni/mcdc/nest/bitmapaddr

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/mcdcstate' into users/chapuni/mcdc/nest/bitmapaddr

chapuni commented:
@evodius96 I think you would be the best person to review my remaining requests.

evodius96 commented:
I agree; I apologize for my silence as I have had several distractions. I plan to start looking at more of these next week.

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/nextcount

Conflicts:
	clang/lib/CodeGen/CodeGenFunction.h

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/loop

Conflicts:
	clang/lib/CodeGen/CGStmt.cpp
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/condop

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/if

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/switch

chapuni committed:
Merge branches 'users/chapuni/cov/single/loop', 'users/chapuni/cov/single/condop' and 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

chapuni committed:
Merge branches 'users/chapuni/cov/single/binop' and 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/expect

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/lnot

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/mcdcstate

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/mcdcstate' into users/chapuni/mcdc/nest/logopstack

chapuni committed:
clangCodeGen: Follow #142155

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/mcdcstate' into users/chapuni/mcdc/nest/covgen

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp (#172557)

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/mcdcstate' into users/chapuni/mcdc/nest/bitmapaddr

Conflicts:
	clang/lib/CodeGen/CodeGenFunction.h (#142155)

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/unify

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/region_segment

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/forfile-base

chapuni committed:
Merge branches 'users/chapuni/cov/single/unify' and 'users/chapuni/cov/merge/region_segment' into users/chapuni/cov/merge/forfile-base

chapuni committed:
Merge branch 'users/chapuni/cov/merge/forfile-base' into users/chapuni/cov/merge/forfile

Conflicts:
	llvm/include/llvm/ProfileData/Coverage/CoverageMapping.h

chapuni committed:
Update branch-export-lcov-unify-instances.test (in #135074)

chapuni committed:
Merge branch 'users/chapuni/cov/merge/forfile' into users/chapuni/cov/merge/strategy

Conflicts:
	llvm/include/llvm/ProfileData/Coverage/CoverageMapping.h

chapuni committed:
Update branch-export-lcov-unify-instances.test (in #135074)

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/mcdcsort

chapuni committed:
Update mcdc-export-json.test (in #159119)

chapuni committed:
Merge branches 'users/chapuni/cov/merge/strategy' and 'users/chapuni/cov/merge/mcdcsort' into users/chapuni/cov/merge/merge-mcdc-base

chapuni committed:
Merge branch 'users/chapuni/cov/merge/merge-mcdc-base' into users/chapuni/cov/merge/merge-mcdc

Conflicts:
	llvm/lib/ProfileData/Coverage/CoverageMapping.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/merge/merge-mcdc' into users/chapuni/cov/merge/trunk

chapuni committed:
-EnableSingleByteCoverage

chapuni committed:
Add comments

chapuni committed:
ReleaseNotes.rst

chapuni committed:
emitCounterSetOrIncrement: Simplify.

- s/`TheMap`/`TheCounterPair`/
- Change the type of `Counter` to `ValueOpt`.

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/nextcount

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/nextcount

chapuni committed:
Merge branch 'users/chapuni/cov/single/nextcount' into users/chapuni/cov/single/unify

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/unify

chapuni committed:
Merge branch 'users/chapuni/cov/single/unify' into users/chapuni/cov/merge/mcdcsort

chapuni committed:
Merge branch 'main' into users/chapuni/cov/merge/mcdcsort

chapuni committed:
Merge branch 'users/chapuni/cov/merge/mcdcsort' into users/chapuni/mcdc/nest/mcdcstate

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/mcdcstate

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/mcdcstate' into users/chapuni/mcdc/nest/lnot

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/cov/single/loop

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/cov/single/condop

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/cov/single/if

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/cov/single/switch

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/cov/single/trunk

Conflicts:
	clang/docs/ReleaseNotes.rst

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/mcdc/nest/bitmapaddr

chapuni committed:
fix: LoopFalse should be used.

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/mcdc/nest/expect

Conflicts:
	clang/include/clang/AST/IgnoreExpr.h
	clang/lib/CodeGen/CodeGenFunction.cpp

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/expect

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/mcdc/nest/covgen

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/covgen

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/lnot

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/lnot' into users/chapuni/mcdc/nest/logopstack

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/logopstack

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/logopstack' into users/chapuni/mcdc/nest/expect

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/expect

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/expect' into users/chapuni/mcdc/nest/covgen

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/covgen

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/covgen' into users/chapuni/mcdc/nest/bitmapaddr

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/bitmapaddr

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/bitmapaddr' into users/chapuni/cov/merge/region_segment

chapuni committed:
Merge branch 'users/chapuni/cov/merge/region_segment' into users/chapuni/cov/merge/forfile

Conflicts:
	llvm/test/tools/llvm-cov/branch-templates.test

chapuni committed:
Merge branch 'users/chapuni/cov/merge/forfile' into users/chapuni/cov/merge/strategy

Conflicts:
	llvm/test/tools/llvm-cov/branch-templates.test

chapuni committed:
Merge branch 'users/chapuni/cov/merge/strategy' into users/chapuni/cov/merge/merge-mcdc

chapuni committed:
Merge branch 'users/chapuni/cov/merge/merge-mcdc' into users/chapuni/cov/merge/trunk

chapuni committed:
LLVM_ABI

chapuni committed:
Merge branch 'users/chapuni/cov/merge/strategy' into users/chapuni/cov/merge/merge-mcdc

chapuni committed:
Merge branch 'users/chapuni/cov/merge/merge-mcdc' into users/chapuni/cov/merge/trunk

chapuni committed:
[Coverage] Introduce the predicate `hasSkipCounter(S)`. NFC.

Also, rename s/getIsCounterPair/getExecSkipCounterExistence/.

chapuni committed:
Merge commit 'cead042fd7b57f7bb454d4ffe6f0413658a4b069' into users/chapuni/cov/single/loop

chapuni committed:
Use hasSkipCounter

chapuni committed:
Merge commit 'cead042fd7b57f7bb454d4ffe6f0413658a4b069' into users/chapuni/cov/single/if

chapuni committed:
Use hasSkipCounter

chapuni committed:
Merge commit 'cead042fd7b57f7bb454d4ffe6f0413658a4b069' into users/chapuni/cov/single/switch

chapuni committed:
Use hasSkipCounter

chapuni committed:
Merge branch 'users/chapuni/cov/single/loop' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

Conflicts:
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Use hasSkipCounter

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop' into users/chapuni/cov/single/trunk

Conflicts:
	clang/docs/ReleaseNotes.rst

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branch 'main' into users/chapuni/mcdc/nest/bitmapaddr

chapuni committed:
Merge branch 'users/chapuni/mcdc/nest/bitmapaddr' into users/chapuni/cov/single/condop

Conflicts:
	clang/lib/CodeGen/CGExprScalar.cpp
	clang/lib/CodeGen/CodeGenPGO.cpp
	clang/lib/CodeGen/CoverageMappingGen.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/single/condop' into users/chapuni/cov/single/execskipexists

chapuni committed:
hasSkipCounter

chapuni committed:
Merge branch 'users/chapuni/cov/single/execskipexists' into users/chapuni/cov/single/loop

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.proftext
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.yaml

chapuni committed:
Merge branch 'users/chapuni/cov/single/execskipexists' into users/chapuni/cov/single/if

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.proftext
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.yaml

chapuni committed:
Merge branch 'users/chapuni/cov/single/execskipexists' into users/chapuni/cov/single/switch

chapuni committed:
getSwitchImplicitDefaultCounterPair: Flatten expressions.

I believe the original form was already readable in this context,
but I rewrote it to match the requested style.

chapuni committed:
Merge branch 'users/chapuni/cov/single/execskipexists' into users/chapuni/cov/single/binop-base

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/binop

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/trunk

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/loop

chapuni committed:
Merge branch 'users/chapuni/cov/single/loop' into users/chapuni/cov/single/switch

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.proftext
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.yaml

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/loop

chapuni committed:
Merge branch 'users/chapuni/cov/single/loop' into users/chapuni/cov/single/switch

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/if

Conflicts:
	clang/test/CoverageMapping/single-byte-counters.cpp
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.proftext
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.yaml
	llvm/test/tools/llvm-cov/Inputs/branch-c-general.c
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.proftext
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.yaml

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop-base' into users/chapuni/cov/single/if

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.proftext
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.yaml
	llvm/test/tools/llvm-cov/Inputs/showLineExecutionCounts-single.yaml

chapuni committed:
Merge branch 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.yaml

chapuni committed:
Merge branch 'users/chapuni/cov/single/trunk' into users/chapuni/cov/single/binop

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/branch-c-general-single.yaml

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/switch

chapuni committed:
Merge branch 'users/chapuni/cov/single/switch' into users/chapuni/cov/single/if

chapuni committed:
Merge branch 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop

chapuni committed:
Merge branch 'main' into users/chapuni/cov/single/if

chapuni committed:
Merge branch 'users/chapuni/cov/single/if' into users/chapuni/cov/single/binop

chapuni committed:
Merge branch 'users/chapuni/cov/single/binop' into users/chapuni/cov/merge/region_segment

Conflicts:
	llvm/test/tools/llvm-cov/branch-macros.test

chapuni committed:
Merge branch 'users/chapuni/cov/merge/region_segment' into users/chapuni/cov/merge/forfile

chapuni committed:
Merge branch 'users/chapuni/cov/merge/forfile' into users/chapuni/cov/merge/strategy

Conflicts:
	llvm/test/tools/llvm-cov/Inputs/branch-templates.cpp

chapuni committed:
Merge branch 'users/chapuni/cov/merge/strategy' into users/chapuni/cov/merge/merge-mcdc

chapuni committed:
Merge branch 'users/chapuni/cov/merge/merge-mcdc' into users/chapuni/cov/merge/trunk
